core/common/src/main/java/alluxio/underfs/UnderFileSystem.java

```diff
@@ -98,15 +102,17 @@ void clear() {
     private static class Key {
       private final String mScheme;
       private final String mAuthority;
+      private final Map<String, String> mProperties;
 
-      Key(AlluxioURI uri) {
+      Key(AlluxioURI uri, Map<String, String> properties) {
```

Comment:
I am a little worried that this will be the performance bottleneck. We discussed to use the version of the site property as the hash key. Why did you drop that idea in favor of this one?